cider/exec/plan/substrait/SubstraitPlan.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "SubstraitPlan.h"
+
+namespace cider::plan {
+
+SubstraitPlan::SubstraitPlan(const substrait::Plan& plan) : plan_(plan) {}
```

Comment:
Minor hint: IMO, better to call it ```SubstraitPlanUtils``` and make it static accepting substrait plan as parameter. Alternatively, caching those results as ```containsAgg``` or ```containsJoin``` locally avoiding further visit.

Reply:
caching result is make sense

Reply:
caching result is make sense, but our substrait plan only have one root , so it seems we can't get any benefit  by caching the result